lib/menu.rb

```diff
@@ -0,0 +1,22 @@
+class Menu
+  attr_reader :menu
+
+  def initialize
+    @menu = {
+      "Chicken burger" => 10,
+      "Beef burger" => 12,
+      "Fish dog" => 11,
+      "Halloumi fritter burger" => 8,
+      "Sloppy Joe dog" => 9,
+      "Truffle fries" => 4,
+      "Mac & Cheese" => 5,
+      "Corn ribs" => 4,
+    }
+  end
+
+  def printed_menu
```

Comment:
For your question about why this method prints "twice" when run in IRB, that's because IRB will print out the returned value of the method you're calling. So two things end being printed:
 * what is printed by `puts`
 * the return value of `printed_menu` - since calling `.each` will return the value of the array it's called on (after executing the block for each item), you end up having the `menu` array returned by the whole method

Reply:
Maybe a way to make this "cleaner" could be to build a return string to return, rather than directly printing using `puts`:
```ruby
menu.each do |dish, price|
      output += "#{dish}: £#{price}"
 end
```

You can then simply decide to "print" (puts) this value or do something else with it, but that will be the job of the caller, rather than of this method